macos-13-conda-forge-standard: linbox fails to build

```
  [linbox-1.7.0+sage-2024-05-18]   [spkg-install] In file included from diophantine-solver.C:26:
  [linbox-1.7.0+sage-2024-05-18]   [spkg-install] In file included from ../../linbox/algorithms/diophantine-solver.h:29:
  [linbox-1.7.0+sage-2024-05-18]   [spkg-install] In file included from ../../linbox/algorithms/rational-solver.h:41:
  [linbox-1.7.0+sage-2024-05-18]   [spkg-install] In file included from ../../linbox/solutions/methods.h:26:
  [linbox-1.7.0+sage-2024-05-18]   [spkg-install] In file included from ../../linbox/matrix/dense-matrix.h:37:
  [linbox-1.7.0+sage-2024-05-18]   [spkg-install] ../../linbox/vector/vector-traits.h:153:15: error: no template named 'binary_function' in namespace 'std'; did you mean '__binary_function'?
  [linbox-1.7.0+sage-2024-05-18]   [spkg-install]   153 |                 public std::binary_function<const std::pair<size_t, Element>&, const std::pair<size_t, Element>&, bool > {
  [linbox-1.7.0+sage-2024-05-18]   [spkg-install]       |                        ~~~~~^~~~~~~~~~~~~~~
  [linbox-1.7.0+sage-2024-05-18]   [spkg-install]       |                             __binary_function
 ```
https://github.com/passagemath/passagemath/actions/runs/14890759836/job/41821968069#step:10:2962

--- a/build/pkgs/linbox/spkg-install.in
+++ b/build/pkgs/linbox/spkg-install.in
@@ -9,6 +9,11 @@ cd src/
 # disabled.  Therefore, disable it always.
 export CPPFLAGS="$CPPFLAGS -DDISABLE_COMMENTATOR"
 
+# From https://github.com/conda-forge/linbox-feedstock/blob/main/recipe/build.sh#L13C1-L16C1
+# givaro defined bool_constant but since C++17 there is also std::bool_constant which leads to ambiguity.
+# linbox needs std::binary_function which was removed in C++17.
+export CXXFLAGS="$CXXFLAGS -std=c++14"
+
 ###############################################################################
 # Configure, build and install LinBox:
 ###############################################################################

